### PR TITLE
Charlesmchen/change media preview sizes

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk 

screen shot 2017-10-24 at 1 25 11 pm
screen shot 2017-10-24 at 1 25 02 pm
screen shot 2017-10-24 at 1 24 54 pm
screen shot 2017-10-24 at 1 24 43 pm

### CODE REVIEW

File: Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m [hunk 1]
```diff
@@ -241,7 +241,7 @@ - (void)ensureContentConstraints
             [self.attachmentView autoPinEdgeToSuperviewEdge:ALEdgeTop withInset:textViewVInset],
             [self.attachmentView autoPinEdgeToSuperviewEdge:ALEdgeBottom withInset:textViewVInset],
             [self.attachmentView autoPinEdgeToSuperviewEdge:ALEdgeLeft withInset:contentHInset],
-            [self.attachmentView autoSetDimension:ALDimensionHeight toSize:300.f],
+            [self.attachmentView autoSetDimension:ALDimensionHeight toSize:150.f],
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m]: Height of attachment preview is pretty subjective.  WDYT?

File: Signal/src/ViewControllers/MediaMessageView.swift [hunk 2]
```diff
@@ -5,24 +5,30 @@
 import Foundation
 import MediaPlayer
 
+@objc
+enum MediaMessageViewMode: UInt {
+    case large
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/MediaMessageView.swift]: We don't use the large mode anymore, but we could.

File: Signal/src/ViewControllers/MessageMetadataViewController.swift [hunk 3]
```diff
@@ -382,7 +382,7 @@ class MessageMetadataViewController: OWSViewController {
         let contentType = attachment.contentType
         if let dataUTI = MIMETypeUtil.utiType(forMIMEType: contentType) {
             let attachment = SignalAttachment(dataSource: dataSource, dataUTI: dataUTI)
-            let mediaMessageView = MediaMessageView(attachment: attachment)
+            let mediaMessageView = MediaMessageView(attachment: attachment, mode: .small)
```
Comment (charlesmchen) [inline Signal/src/ViewControllers/MessageMetadataViewController.swift]: IMO small mode works better here.  WDYT?

File: Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m (hunk 1 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/ConversationView/ConversationInputToolbar.m]: I built locally - this feels reasonable.

File: Signal/src/ViewControllers/FullImageViewController.m [hunk 4]
```diff
@@ -77,14 +78,65 @@ - (instancetype)initWithAttachment:(TSAttachmentStream *)attachmentStream
...
+    } else if (self.attachment) {
+        return self.attachment.dataUrl;
+    } else {
+        return nil;
+    }
+}
+
+- (NSData *)fileData
+{
+    if (_fileData) {
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/FullImageViewController.m]:
```
 _____________________________________
/ It looks like you're trying to do   \
| memoization. Did you mean to invert |
\ this condition? e.g. !_fileData     /
 -------------------------------------
 \
  \
     __
    /  \
    |  |
    @  @
    |  |
    || |/
    || ||
    |\_/|
    \___/
```

File: Signal/src/ViewControllers/MediaMessageView.swift [hunk 5]
```diff
@@ -36,15 +42,17 @@ class MediaMessageView: UIView, OWSAudioAttachmentPlayerDelegate {
 
     @available(*, unavailable, message:"use attachment: constructor instead.")
     required init?(coder aDecoder: NSCoder) {
+        self.mode = .large
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/MediaMessageView.swift]: TIOLI: You can use `fatalError("uninimplemented")` and stop worrying about stubbing bogus data in methods we don't use.

File: Signal/src/ViewControllers/MediaMessageView.swift [hunk 6]
```diff
@@ -200,7 +215,10 @@ class MediaMessageView: UIView, OWSAudioAttachmentPlayerDelegate {
         let aspectRatio = image.size.width / image.size.height
         addSubviewWithScaleAspectFitLayout(view:imageView, aspectRatio:aspectRatio)
         contentView = imageView
-    }
+
+        imageView.isUserInteractionEnabled = true
+        imageView.addGestureRecognizer(UITapGestureRecognizer(target:self, action:#selector(imageTapped)))
+}
```
Comment (michaelkirk) [inline Signal/src/ViewControllers/MediaMessageView.swift]: weird indent

File: Signal/src/ViewControllers/MessageMetadataViewController.swift (hunk 3 above)
Comment (michaelkirk) [reply Signal/src/ViewControllers/MessageMetadataViewController.swift]: 👍

**Comment by charlesmchen** [comment]
> Also, when I tap on a GIF I see an empty view controller.

Fixed!

File: Signal/src/ViewControllers/FullImageViewController.m (hunk 4 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/FullImageViewController.m]: Done. 

Clippppppy!

File: Signal/src/ViewControllers/MediaMessageView.swift (hunk 5 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/MediaMessageView.swift]: Nice!

File: Signal/src/ViewControllers/MediaMessageView.swift (hunk 6 above)
Comment (charlesmchen) [reply Signal/src/ViewControllers/MediaMessageView.swift]: Done.

**Pull request merged**